src/main/java/org/apache/sling/discovery/oak/cluster/ClusterReader.java

```diff
@@ -0,0 +1,142 @@
+        if (slingId==null) {
+            throw new IllegalStateException("slingId must not be null");
+        }
+        final String myClusterNodePath = config.getClusterInstancesPath()+"/"+slingId;
+        // SLING-6924 case 1 : /var/discovery/oak/clusterInstances/<slingId> can be non existant == null
+        final Resource myClusterNode = resourceResolver.getResource(myClusterNodePath);
+        if (myClusterNode == null) {
+            // SLING-6924 : return null case 1
+            return null;
+        }
+        ValueMap resourceMap = myClusterNode.adaptTo(ValueMap.class);
+        // SLING-6924 case 2 : /var/discovery/oak/clusterInstances/<slingId> can exist BUT leaderElectionId not yet set
+        //    namely the "leaderElectionId" is only written when resetLeaderElectionId() is called - which happens
+        //    on OakViewChecker.activate (or when isolated) - and this activate *can* happen after properties
+        //    or announcements have been written - those end up below /var/discovery/oak/clusterInstances/<slingId>/
+        String result = resourceMap.get("leaderElectionId", String.class);
+
+        // SLING-6924 : return null case 2 (if leaderElectionId is indeed null, that is)
+        return result;
+    }
+
+    /**
+     * Reads all discovery.oak related data for clusterNodeId
+     * and fills the internal structure with the resulting info -
+     * or remembers that the clusterNode was incomplete (partially started)
+     * @param clusterNodeId the clusterNodeId to be read
+     * @return the instance if it is has all required data stored (which does not
+     * necessarily mean it has completely started - syncToken is not checked at this stage yet)
+     * @throws PersistenceException 
+     */
+    public InstanceReadResult readInstance(int clusterNodeId, boolean failOnMissingSyncToken) throws PersistenceException {
+        final String slingId = idMapService.toSlingId(clusterNodeId, resourceResolver);
+        if (slingId == null) {
+            idMapService.clearCache();
+            return InstanceReadResult.fromErrorMsg("no slingId mapped for clusterNodeId=" + clusterNodeId);
+        }
+        final String leaderElectionId = readLeaderElectionId(slingId);
+        if (leaderElectionId == null) {
+            return InstanceReadResult.fromErrorMsg("no leaderElectionId available yet for slingId=" + slingId);
+        }
+        final boolean hasSeenLocalInstance = hasSeenLocalinstance(clusterNodeId, slingId, leaderElectionId);
+        final long syncToken = readSyncToken(slingId);
+        if (syncToken == -1 && failOnMissingSyncToken && !hasSeenLocalInstance) {
+            return InstanceReadResult.fromErrorMsg("no syncToken available yet for slingId=" + slingId);
+        }
+        return InstanceReadResult.fromInstance(new InstanceInfo(clusterNodeId, slingId, syncToken, leaderElectionId));
+    }
+
+    private boolean hasSeenLocalinstance(int clusterNodeId, String slingId,
```

Comment:
Minor typo. I would rather write `hasSeenLocalInstance` instead. Or did you deliberately use a lower case `i` to better distinguish it from the preceding `l`?